services/QuillLMS/engines/evidence/lib/evidence/evidence/synthetic/initial_data.rb

```diff
@@ -0,0 +1,138 @@
+# frozen_string_literal: true
+
+module Evidence
+  module Synthetic
+    class InitialData
+      Result = Struct.new(:text, :seed, keyword_init: true)
+
+      WORD_SPLIT_COUNT = 70
+      SPACE = ' '
+      BLANK = ''
+      PERIOD = '.'
+
+      FULL_COUNT = 100
+      FULL_NOUN_COUNT = 50
+      SECTION_COUNT = 50
+
+      TEMP_PASSAGE = 1
+      TEMP_SECTION = 0.5 # give a lower temp (creativity) when it has less info
+
+      attr_reader :passage, :stem, :nouns, :results
+
+      # returns a hash of the form {'csv name' => CSVString, 'csv name2' =>...}
+      def self.generate_csvs(activity_id:, nouns: [])
+        activity = Activity.find(activity_id)
+        passage = activity.passages.first.text
+        prompts = activity.prompts
+        short_name = activity.title.first(20).gsub(' ', '_')
+
+        csvs = {}
+
+        prompts.each.with_index do |prompt, index|
```

Comment:
It's equivalent code. This is just a style thing, but I do like the composable `.with_index` since you can use it with anything, e.g. you can do `map.with_index` but there is no `map_with_index`